packages/core/src/module-resolver.ts

```diff
@@ -421,6 +422,32 @@ export class Resolver {
     }
   }
 
+  private handleImplicitTestScripts<R extends ModuleRequest>(request: R): R {
+    //TODO move the extra forwardslash handling out into the vite plugin
+    const candidates = [
+      '@embroider/core/test-support.js',
+      '/@embroider/core/test-support.js',
+      './@embroider/core/test-support.js',
+    ];
+
+    if (!candidates.includes(request.specifier)) {
+      return request;
+    }
+
+    let pkg = this.packageCache.ownerOfFile(request.fromFile);
+    if (pkg?.root !== this.options.engines[0].root) {
+      throw new Error(
+        `bug: found an import of ${request.specifier} in ${request.fromFile}, but this is not the top-level Ember app. The top-level Ember app is the only one that has support for @embroider/core/vendor. If you think something should be fixed in Embroider, please open an issue on https://github.com/embroider-build/embroider/issues.`
+      );
+    }
+
+    if (!pkg?.isV2Ember()) {
+      throw new Error(`bug: an import of ${request.specifier} in non-ember package at ${request.fromFile}`);
+    }
```

Comment:
We can remove this check, the check up above for being the first engine is sufficient.